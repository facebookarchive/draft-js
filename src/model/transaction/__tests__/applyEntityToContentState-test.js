/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 * @oncall draft_js
 */

'use strict';
import type ContentState from 'ContentState';

const SelectionState = require('SelectionState');

const applyEntityToContentState = require('applyEntityToContentState');
const getSampleStateForTesting = require('getSampleStateForTesting');

const {contentState, selectionState} = getSampleStateForTesting();

const initialBlock = contentState.getBlockMap().first();
const secondBlock = contentState.getBlockAfter(initialBlock.getKey());

const selectBlock = new SelectionState({
  anchorKey: initialBlock.getKey(),
  anchorOffset: 0,
  focusKey: initialBlock.getKey(),
  focusOffset: initialBlock.getLength(),
});

const selectAdjacentBlocks = new SelectionState({
  anchorKey: initialBlock.getKey(),
  anchorOffset: 0,
  focusKey: secondBlock?.getKey(),
  focusOffset: secondBlock?.getLength(),
});

const assertApplyEntityToContentState = (
  entityKey: null | $TEMPORARY$string<'x'>,
  selection: $FlowFixMe | SelectionState = selectionState,
  content: ContentState = contentState,
) => {
  expect(
    applyEntityToContentState(content, selection, entityKey)
      .getBlockMap()
      .toJS(),
  ).toMatchSnapshot();
};

test('must apply entity key', () => {
  assertApplyEntityToContentState('x', selectBlock);
});

test('must apply null entity', () => {
  assertApplyEntityToContentState(null, selectBlock);
});

test('must apply entity key accross multiple blocks', () => {
  assertApplyEntityToContentState('x', selectAdjacentBlocks);
});

test('must apply null entity key accross multiple blocks', () => {
  assertApplyEntityToContentState(null, selectAdjacentBlocks);
});
