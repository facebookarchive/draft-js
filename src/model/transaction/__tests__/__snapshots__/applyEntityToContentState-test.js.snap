// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must apply entity key 1`] = `
Object {
  "a": Object {
    "characterList": Array [
      Object {
        "entity": "x",
        "style": Array [],
      },
      Object {
        "entity": "x",
        "style": Array [],
      },
      Object {
        "entity": "x",
        "style": Array [],
      },
      Object {
        "entity": "x",
        "style": Array [],
      },
      Object {
        "entity": "x",
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "a",
    "text": "Alpha",
    "type": "unstyled",
  },
  "b": Object {
    "characterList": Array [
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "b",
    "text": "Bravo",
    "type": "unordered-list-item",
  },
  "c": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "c",
    "text": "Test",
    "type": "code-block",
  },
  "d": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "d",
    "text": "",
    "type": "code-block",
  },
  "e": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "e",
    "text": "",
    "type": "code-block",
  },
  "f": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "f",
    "text": "Charlie",
    "type": "blockquote",
  },
}
`;

exports[`must apply entity key accross multiple blocks 1`] = `
Object {
  "a": Object {
    "characterList": Array [
      Object {
        "entity": "x",
        "style": Array [],
      },
      Object {
        "entity": "x",
        "style": Array [],
      },
      Object {
        "entity": "x",
        "style": Array [],
      },
      Object {
        "entity": "x",
        "style": Array [],
      },
      Object {
        "entity": "x",
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "a",
    "text": "Alpha",
    "type": "unstyled",
  },
  "b": Object {
    "characterList": Array [
      Object {
        "entity": "x",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "x",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "x",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "x",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "x",
        "style": Array [
          "BOLD",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "b",
    "text": "Bravo",
    "type": "unordered-list-item",
  },
  "c": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "c",
    "text": "Test",
    "type": "code-block",
  },
  "d": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "d",
    "text": "",
    "type": "code-block",
  },
  "e": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "e",
    "text": "",
    "type": "code-block",
  },
  "f": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "f",
    "text": "Charlie",
    "type": "blockquote",
  },
}
`;

exports[`must apply null entity 1`] = `
Object {
  "a": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "a",
    "text": "Alpha",
    "type": "unstyled",
  },
  "b": Object {
    "characterList": Array [
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "b",
    "text": "Bravo",
    "type": "unordered-list-item",
  },
  "c": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "c",
    "text": "Test",
    "type": "code-block",
  },
  "d": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "d",
    "text": "",
    "type": "code-block",
  },
  "e": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "e",
    "text": "",
    "type": "code-block",
  },
  "f": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "f",
    "text": "Charlie",
    "type": "blockquote",
  },
}
`;

exports[`must apply null entity key accross multiple blocks 1`] = `
Object {
  "a": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "a",
    "text": "Alpha",
    "type": "unstyled",
  },
  "b": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "b",
    "text": "Bravo",
    "type": "unordered-list-item",
  },
  "c": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "c",
    "text": "Test",
    "type": "code-block",
  },
  "d": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "d",
    "text": "",
    "type": "code-block",
  },
  "e": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "e",
    "text": "",
    "type": "code-block",
  },
  "f": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "f",
    "text": "Charlie",
    "type": "blockquote",
  },
}
`;
